tests/integration/components/enterprise-banner-test.js

```diff
@@ -0,0 +1,157 @@
+    assert.ok(this.$('.enterprise-banner-trial').text().match(/Your trial license has expired/));
+    assert.notOk(this.$('.enterprise-banner-trial').hasClass('alert'));
+  });
+});
+
+test('renders paid banner 60 days from expiry', function (assert) {
+  this.server.get('/v3/enterprise_license', (schema, response) => {
+    return {
+      'license_id': 'ad12345',
+      'seats': '30',
+      'active_users': '21',
+      'license_type': 'paid',
+      'expiration_time': new Date(new Date().getTime() + 1000 * 60 * 60 * 24 * 59).toISOString()
+    };
+  });
+  assert.expect(3);
+
+  this.render(hbs`{{enterprise-banner}}`);
+
+  wait().then(() => {
+    assert.ok(this.$('.enterprise-banner-license').text().match(/Your license expires 2 months from now/));
+    assert.ok(this.$('.enterprise-banner-license').hasClass('alert'));
+    assert.ok(this.$('.enterprise-banner-license button').length);
+  });
+});
+
+test('renders paid banner 30 days from expiry', function (assert) {
+  this.server.get('/v3/enterprise_license', (schema, response) => {
+    return {
+      'license_id': 'ad12345',
+      'seats': '30',
+      'active_users': '21',
+      'license_type': 'paid',
+      'expiration_time': new Date(new Date().getTime() + 1000 * 60 * 60 * 24 * 26).toISOString()
+    };
+  });
+  assert.expect(3);
+
+  this.render(hbs`{{enterprise-banner}}`);
+
+  wait().then(() => {
+    assert.ok(this.$('.enterprise-banner-license').text().match(/Your license expires 26 days from now/));
+    assert.ok(this.$('.enterprise-banner-license').hasClass('alert'));
+    assert.ok(this.$('.enterprise-banner-license button').length);
+  });
+});
+
+test('renders paid banner 10 days from expiry', function (assert) {
+  this.server.get('/v3/enterprise_license', (schema, response) => {
+    return {
+      'license_id': 'ad12345',
+      'seats': '30',
+      'active_users': '21',
+      'license_type': 'paid',
+      'expiration_time': new Date(new Date().getTime() + 1000 * 60 * 60 * 24 * 8).toISOString()
+    };
+  });
+  assert.expect(3);
+
+  this.render(hbs`{{enterprise-banner}}`);
+
+  wait().then(() => {
+    assert.ok(this.$('.enterprise-banner-license').text().match(/Your license expires 8 days from now/));
+    assert.ok(this.$('.enterprise-banner-license').hasClass('alert'));
+    assert.notOk(this.$('.enterprise-banner-license button').length);
+  });
+});
+
+test('renders seats banner nearing excess', function (assert) {
+  this.server.get('/v3/enterprise_license', (schema, response) => {
+    return {
+      'license_id': 'ad12345',
+      'seats': '30',
+      'active_users': '26',
+      'license_type': 'paid',
+      'expiration_time': new Date(new Date().getTime() + 1000 * 60 * 60 * 24 * 8).toISOString()
+    };
+  });
+  assert.expect(2);
+
+  this.render(hbs`{{enterprise-banner}}`);
+
+  wait().then(() => {
+    assert.ok(this.$('.enterprise-banner-seats').text().match(/You’re approaching the maximum seats that your license permits/));
```

Comment:
v nice to see the typographer apostrophe here 😍

Reply:
Do we get a prize?